File: HWLesson5/app/src/main/java/com/example/hwlesson5/MainActivity.java
package com.example.hwlesson5;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    private EditText editTextName;
    private TextView textViewResult;
    private ActivityResultLauncher<Intent> activityResultLauncher;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editTextName = findViewById(R.id.editTextName);
        textViewResult = findViewById(R.id.textViewResult);
        Button btnProceed = findViewById(R.id.btnProceed);

        activityResultLauncher = registerForActivityResult(
                new ActivityResultContracts.StartActivityForResult(),
                result -> {
                    if (result.getResultCode() == RESULT_OK && result.getData() != null) {
                        String modifiedName = result.getData().getStringExtra("MODIFIED_NAME");
                        textViewResult.setText(modifiedName);
                    }
                });

        btnProceed.setOnClickListener(v -> {
            String name = editTextName.getText().toString();
            Intent intent = new Intent(MainActivity.this, SecondActivity.class);
            intent.putExtra("USER_NAME", name);
            activityResultLauncher.launch(intent);
        });
    }
}

File: HWLesson24_orig/app/src/main/java/com/example/hwlesson24/MainActivity.kt
package com.example.hwlesson24

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val textView = TextView(this)
        textView.text = "\n\n\n\t\tAPI_URL: ${BuildConfig.API_URL}\n\t\tApp Name: ${getString(R.string.app_name)}"
        setContentView(textView)
    }
}
